WordPress/gutenberg
# Sidebar: Add slot above tabs for items applying to block as a whole

Related:
- https://github.com/WordPress/gutenberg/pull/45005

## What?

Adds a new inspector controls group slot for blocks to be able to render items above the sidebar tabs and uses this to move the Query block's "create a new post" link there.

## Why?

The Query block's "create a new post" link doesn't make a lot of sense within the sidebar tabs. It has previously relied upon appearing as though it were part of the Block Card.

## How?

- Adds new InspectorControls group called `BlockInformation`
- Renders the Query block's "create a new post" link into the above slot

## Questions / Notes

This new slot might be open to blocks or 3rd parties abusing it to promote controls above the sidebar tabs where it should be for things that apply to the block as a whole and don't make sense within the tabs.

Do the benefits in still allowing something like the Query block's "create a new post" link outweigh the issues here?

Is there an alternative location where the link might fit better? 

It appears from the [original issue](https://github.com/WordPress/gutenberg/issues/23461) there were limited options, and its current home was more of a quick win until something better came along.

## Testing Instructions

1. Edit a post, add a query block, and select it
2. In the sidebar, confirm the "create a new post" link is above the appearance and settings tabs.

## Screenshots or screencast 

| Before | After |
|---|---|
| <img width="278" alt="Screenshot 2022-11-01 at 5 40 31 pm" src="https://user-images.githubusercontent.com/60436221/199184281-ab450ad0-a7f4-4b51-b9b3-f8befe0fdc40.png"> | <img width="278" alt="Screenshot 2022-11-01 at 5 41 03 pm" src="https://user-images.githubusercontent.com/60436221/199184294-681954ef-49e7-483b-b2e0-dc13d0f67b80.png"> |
